toqito/matrices/operator_sinkhorn.py

```diff
@@ -0,0 +1,199 @@
+"""Sinkhorn operator."""
+
+import warnings
+
+import numpy as np
+import scipy as sp
+
+from toqito.channels.partial_trace import partial_trace
+from toqito.matrix_props.is_square import is_square
+
+
+def operator_sinkhorn(
+    rho: np.ndarray, dim: list[int] = None, tol: float = np.sqrt(np.finfo(float).eps), max_iterations: int = 10_000
+) -> tuple[np.ndarray, list[np.ndarray]]:
+    r"""Perform operator Sinkhorn iteration (iterative Sinkhorn algorithm) on a quantum system's density matrix.
+
+    This function is adapted from QETLAB. :cite:`QETLAB_link`.
+
+    This function relies on Sinkhorn's theorem :cite:`Sinkhorn_1964_Relationship` which states "for any
+    positive-definite square matrix, there exist diagonal matrices :math:`D_1` and :math:`D_2` such that
+    :math:`D_1 \cdot A \cdot D_2` is doubly stochastic.
+
+    The iterative Sinkhorn algorithm alternately rescales the input density matrix of the quantum system along each
+    subsystem so that its marginals become uniform (i.e., partial traces or row/column sums become proportional to
+    identity), until convergence.
+
+    Upon convergence we end up with a density matrix `sigma` (:math:`\sigma`), which is locally equivalent to the given
+    multipartite density matrix `rho` (:math:`\rho`), but having maximally mixed subsystems. The algorithm will also
+    return the list of filtering operations which can be directly applied on :math:`\rho` to arrive at :math:`\sigma`.
+    Such converted forms of density matrices are useful to analyse entangled systems and to study effects of operations
+    on each subsystems (as discussed in :cite:`Gurvits_2004_Classical`).
+
+    Examples
+    =======
+
+    To obtain a locally normalized density matrix of a random 2-qubit bipartite state using the Sinkhorn operator,
+     consider the following example:
+
+    ..jupyter-execute::
+        import numpy as np
+        from toqito.matrices import operator_sinkhorn
+        from toqito.rand import random_density_matrix
+
+        rho = random_density_matrix(4, seed=42)
+        print(f"{rho=}")
+
+        sigma, local_ops = operator_sinkhorn(rho=rho_random, dim=[2, 2])
+        print(f"{sigma=}")
+        print(f"{local_ops[0]=}")
```

Comment:
```suggestion
    ..jupyter-execute::
    
        import numpy as np
        from toqito.matrices import operator_sinkhorn
        from toqito.rand import random_density_matrix

        rho = random_density_matrix(4, seed=42)
        print(f"{rho=}")

        sigma, local_ops = operator_sinkhorn(rho=rho_random, dim=[2, 2])
        print(f"{sigma=}")
        print(f"{local_ops[0]=}")
```
Not working as expected. 
![image](https://github.com/user-attachments/assets/15c7e3cc-a7b9-4457-a471-4814e29f59f9)

Reply:
@purva-thakre thanks for this, I just learned how to compile and check the docs. This should be fixed now.